src/Service/PackageSynchronizer/ComposerPackageSynchronizer.php

```diff
@@ -54,7 +56,16 @@ public function synchronize(Package $package): void
                 }
             }
 
+            if (!$latest) {
+                throw new \RuntimeException('Package not found');
+            }
+
             $name = $latest->getPrettyName();
+
+            if (!preg_match(self::NAME_PATTERN, $name, $matches) || empty($matches)) {
```

Comment:
i think we can move this also to `Package` and protect domain from setting incorrect package name